lib/child_process.js

```diff
@@ -611,35 +617,35 @@ function normalizeSpawnArguments(file, args, options) {
 
   if (options.shell) {
     validateArgumentNullCheck(options.shell, 'options.shell');
-    const command = ArrayPrototypeJoin([file, ...args], ' ');
+    const fullCommand = ArrayPrototypeJoin([command, ...args], ' ');
     // Set the shell, switches, and commands.
     if (process.platform === 'win32') {
       if (typeof options.shell === 'string')
-        file = options.shell;
+        command = options.shell;
```

Comment:
I think this change makes the code more confusing, IMO it makes more sense to keep `file` for the shell executable and `command` for the actual command.

Reply:
this is what I mean:

https://nodejs.org/api/child_process.html#child_processspawncommand-args-options

in the documentation, the first parameter is called "command" (which makes sense), while in the code it is called "file" , which is confusing.

The initial commit I did which I just reverted, also changed the name of the parameters in the code to match the proper name used in the documentation.

Reply:
I'm not sure "command" is much better than "file" tbh. According to https://stackoverflow.com/a/2051031, ISO C11 refers to it as the "program name". I would call it "child process argv0".